### DISCUSSION TITLE
Creating custom data source

### DISCUSSION THREAD

**Discussion by Hugo-Persson** [ghostfolio/ghostfolio#1917] [Q&A] [answered]
Hello, is there any way of creating a custom data source. Yahoo doesn't expose data for the items I have in my portfolio. A lot of my stocks and Mutual funds are Swedish and Yahoo doesn't find them, even by ISIN.

**Comment by dtslvr** [comment]
Hello @Hugo-Persson

Can you please provide an example of a stock (with ISIN) you are trying to add?

In the self-hosting variant, there is the concept of a `MANUAL` data source, where you could manually add the records to the `MarketData` database table.

**Reply by Hugo-Persson** [reply]
Hi @dtslvr, 

The ISIN I am trying to add is `SE0002793943`, how do I use MANAUL data source? Do there exist some documentation on it?

**Reply by Hugo-Persson** [reply]
@dtslvr do you have any updates on this?

**Reply by dtslvr** [reply, marked as answer]
Here is a quick outline of how to accomplish it:

1. Connect to the database
2. Add an entry in the table `SymbolProfile` with `dataSource` = `MANUAL` and `symbol` = `SE0002793943`
3. Add an entry in the table `Order` linking to the created `SymbolProfile`
4. Add entries to the table `MarketData`

With [Ghostfolio Premium](https://ghostfol.io/en/pricing) `SE0002793943` is automatically supported.

**Reply by Hugo-Persson** [reply]
Thank you so much, I think I got an outline for adding my own data
